src/main/resources/templates/home.html

```diff
@@ -9,7 +9,7 @@
             <li class="list-group-item" th:each="submission : ${submissionPosts}">
                 <div class="panel panel-default">
                     <div class="panel-body">
-                        <a th:href="@{/posts/__${submission.title}__}" th:text="${submission.title}">Title ...</a>
+                        <a th:href="@{/posts/__${submission.id}__}" th:text="${submission.title}">Title ...</a>
```

Comment:
Using the ID instead of the title for the URL of the AMA is a good way to get around special character in the title. Good Job :100: